compiler/src/iree/compiler/Codegen/LLVMCPU/KernelDispatch.cpp

```diff
@@ -2039,6 +1999,67 @@ static LogicalResult adjustTileSizesForUnPackOp(func::FuncOp entryPointFn,
                                                tileSizesList, pipeline);
 }
 
+static void setLoweringConfigForComputeOps(func::FuncOp entryPointFn,
+                                           ArrayRef<Operation *> computeOps,
+                                           Operation *rootOperation) {
+  auto ctx = entryPointFn.getContext();
+  auto rootLoweringConfig = getLoweringConfig(rootOperation);
+  auto distTileSizes = rootLoweringConfig.getTileSizeVals(0);
+  auto tileAndFuseSizes = rootLoweringConfig.getTileSizeVals(1);
+  auto targetAttr = IREE::HAL::ExecutableTargetAttr::lookup(entryPointFn);
+  auto targetMLTransInfo =
+      TargetMLTransformInfo::getTargetMLTransformInfo(targetAttr);
+  for (auto op : computeOps) {
+    if (op == rootOperation) continue;
+    int numLoops = cast<TilingInterface>(op).getLoopIteratorTypes().size();
+    TileSizesListType tileSizesList = {distTileSizes, tileAndFuseSizes};
+    SmallVector<int64_t> zeros(tileAndFuseSizes.size(), 0);
+    TypeSwitch<Operation *>(op)
+        .Case<tensor::PackOp>([&](auto packOp) {
+          SmallVector<int64_t> vecTileSizes =
+              getPackVectorTileSizes(entryPointFn, packOp);
+          tileSizesList.push_back(zeros);
+          tileSizesList.push_back(vecTileSizes);
+          for (auto &ts : tileSizesList) ts.resize(numLoops, 0);
+          auto packLoweringConfig =
+              IREE::Codegen::LoweringConfigAttr::get(ctx, tileSizesList);
+          setLoweringConfig(op, packLoweringConfig);
+        })
+        .Case<linalg::GenericOp>([&](auto genericOp) {
```

Comment:
It might be better to move all of this out of the `Case` statement itself into a function, and document exactly what is happening here. This is fairly non-trivial?

Reply:
yep, that would be great. There is also some code that is common and could be refactored.

Reply:
I will do it after I address other concerns

Reply:
The common part is ~10 lines here, and the refactoring would have more than 100 lines changes. Because there are three generic op configuration methods. I will do it as a follow-up, so we can have less changes this PR. (it's already quite big.)